Tools/autotest/ardusub.py

```diff
@@ -164,6 +164,50 @@ def test_gripper_mission(self):
         if ex is not None:
             raise ex
 
+    def dive_set_position_target(self):
+        self.change_mode('GUIDED')
+        self.wait_ready_to_arm()
+        self.arm_vehicle()
+
+        startpos = self.mav.recv_match(type='GLOBAL_POSITION_INT',
+                                       blocking=True)
+
+        lat = 5
+        lon = 5
+        alt = 10
+
+        tstart = self.get_sim_time()
+        while True:
+            if self.get_sim_time() - tstart > 200:
+                raise NotAchievedException("Did not move far enough")
+            # send a position-control command
+            self.mav.mav.set_position_target_global_int_send(
```

Comment:
Should we be sending this repeatedly, or just once before the loop?

Reply:
guided mode times movements out if it doesn't receive stuff often enough (GUIDED_POSVEL_TIMEOUT_MS).